src/app/space/create/pipelines/services/pipelines.service.ts

```diff
@@ -62,6 +67,22 @@ export class PipelinesService {
     this.apiUrl = witUrl + 'user/services';
   }
 
+  getCurrentPipelines(): Observable<BuildConfig[]> {
+    return this.runtimePipelinesService.current
+      .distinctUntilChanged()
+      .combineLatest(
+        this.getOpenshiftConsoleUrl(),
+        this.setupBuildConfigLinks);
```

Comment:
Would it make sense to add the `.share()` operator to the Observable as returned by these two functions, rather than doing it at the component callsite?

Also, it shouldn't make any real difference since `getOpenShiftConsoleUrl` will only emit once, but I think it would read easier if `distinctUntilChanged` was after `combineLatest`.

Reply:
I was reluctant to `share()` in the service since I felt like it was consumer-specific behavior. In this case two of the consumers do want to `share`, however the Pipelines page doesn't need it (as far as I understand the situation; I could be wrong).

Reply:
Okay, fair enough.

Reply:
For the `distinctUntilChanged`, I think the placement changes it's behavior. I want to apply `distinctUntilChanged` only to the first observable being combined and not to the combined observable. That way the projection function will be run less often. At least that's my understanding. Thoughts?

Reply:
Hmm, maybe it's a bit confusing and I should convert it to the Observable form
```
Obserable.combineLatest(
  obs1.distinctUntilChanged()
  obs2
  projectionFn
)
```

Reply:
Right, but if one of those combined Observables only emits once, it's running the project function a constant one-fewer time, right? Not really a big performance concern.

Reply:
Yea that could work too. Personally I find that static combineLatest generally just easier to read anyway, and in that form the distinctUntilChanged is pretty clear regardless of whether it's on the first inner Observable or the final one and what's going to happen overall. I still think it should be fine to apply it to the combined Observable though.

Reply:
It will run the project function anytime any one of the 'inner' observables emits. So if the buildConfig emits `[]` 10 times, it will project that with the console url 10 times. No?

Reply:
Ahh yes, I see what you mean now. Yea, that is true.

Reply:
Sorry, I was thinking in terms of the output Observable, and I don't know what I was thinking with the projection function part...